taskt/Core/Automation/Commands/Loop Commands/LoopCollectionCommand.cs

```diff
@@ -1,57 +1,63 @@
-﻿using System;
+﻿using Microsoft.Office.Interop.Outlook;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OpenQA.Selenium;
+using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Xml.Serialization;
 using System.Data;
+using System.Linq;
 using System.Windows.Forms;
-using taskt.UI.Forms;
-using taskt.UI.CustomControls;
-using Microsoft.Office.Interop.Outlook;
+using System.Xml.Serialization;
+using taskt.Core.Automation.Attributes.ClassAttributes;
+using taskt.Core.Automation.Attributes.PropertyAttributes;
+using taskt.Core.Automation.Engine;
+using taskt.Core.Script;
 using taskt.Core.Utilities.CommonUtilities;
+using taskt.UI.CustomControls;
+using taskt.UI.Forms;
 
 namespace taskt.Core.Automation.Commands
 {
     [Serializable]
-    [Attributes.ClassAttributes.Group("Loop Commands")]
-    [Attributes.ClassAttributes.Description("This command allows you to repeat actions several times (loop).  Any 'Begin Loop' command must have a following 'End Loop' command.")]
-    [Attributes.ClassAttributes.UsesDescription("Use this command when you want to iterate over each item in a list, or a series of items.")]
-    [Attributes.ClassAttributes.ImplementationDescription("This command recursively calls the underlying 'BeginLoop' Command to achieve automation.")]
-    public class LoopListCommand : ScriptCommand
+    [Group("Loop Commands")]
+    [Description("This command iterates over a collection to let user perform actions on the collection items.")]
+    public class LoopCollectionCommand : ScriptCommand
     {
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
-        [Attributes.PropertyAttributes.PropertyDescription("Please input the list variable to be looped")]
-        [Attributes.PropertyAttributes.InputSpecification("Enter a variable which contains a list of items")]
-        [Attributes.PropertyAttributes.SampleUsage("[vMyList]")]
-        [Attributes.PropertyAttributes.Remarks("Use this command to iterate over the results of the Split command.")]
+        [PropertyDescription("Input Collection")]
+        [InputSpecification("Provide a collection variable.")]
+        [SampleUsage("{vMyCollection}")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
         public string v_LoopParameter { get; set; }
 
-        public LoopListCommand()
+        [XmlAttribute]
+        [PropertyDescription("Output Collection Item Variable")]
+        [InputSpecification("Select or provide a variable from the variable list.")]
+        [SampleUsage("vUserVariable")]
+        [Remarks("If you have enabled the setting **Create Missing Variables at Runtime** then you are not required" +
```

Comment:
Lets add a note about Datarow usage given the new functionality.

Reply:
done